packages/shared/src/hooks/useIncrementReadingRank.ts

```diff
@@ -16,14 +17,27 @@ export default function useIncrementReadingRank(): ReturnType {
   const { user } = useContext(AuthContext);
   const queryKeyRef = useRef<string[]>();
   const queryClient = useQueryClient();
-  const [clearQueries] = useDebounce(() => {
-    queryClient.invalidateQueries(queryKeyRef.current);
-  }, 100);
+  const [clearQueries] = useDebounce(
+    async (readToday = false): Promise<void> => {
+      const promises = [queryClient.invalidateQueries(queryKeyRef.current)];
+      if (!readToday) {
+        promises.push(
+          queryClient.invalidateQueries(
+            generateQueryKey(RequestKey.UserStreak, user),
+          ),
+        );
+      }
+      await Promise.all(promises);
+    },
+    100,
```

Comment:
Why do we debounce it actually? (product req or engineering thing)

Reply:
I didn't dig deeper here, I just replicated the existing behavior. Will try to understand it now...

Reply:
The setTimeout and clearing of queries was introduced here: https://github.com/dailydotdev/apps/commit/630e6ca4d5799be56d87067314b787a2e09b2857#diff-2ca4b5627b297d8adac0ec5fa73626249c395ceb53225a067d128ed3fad2a6d5

Maybe it was because of animations? 🤔

Reply:
Hmm probably a edge-case before feel free to remove for your needs if we can get away with not having it for the new one.
But guess we support both so then let's leave to avoid it (can't fully recall)